Fix EFS and new Transit Gateway code

jsf9k opened the pull request:
## 🗣 Description

When I tried to deploy to the "env0 (Staging)" account, I ran into several issues with the EFS code and the new Transit Gateway code that uses route tables specific to each assessment environment.

## 💭 Motivation and Context

EFS (persistent storage) and assessment environments that are isolated from one another are both required for the COOL.

## 🧪 Testing

I used these changes to successfully deploy to the "env0 (Staging)" account.

## 🚥 Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

jsf9k committed:
Add the Shared Services account provisioner role as a provider

This provider must be used to manipulate the Transit Gateway route
tables, since those resources live in that account.

jsf9k committed:
Fix some typos/bugs in locals.tf

jsf9k committed:
Update code that finds the proper Images account to use

The logic needed to change since @felddy went through and gave all our
COOL accounts consistent names.

jsf9k committed:
Move locals to locals.tf

The locals needed to change anyway due to the renaming of accounts in
the COOL.

Also fix a few typos where the workspace needs to be hardcoded as
"production" since there is only a single such account.

jsf9k committed:
Add some permissions required for EFS and TGW route table manipulations

jsf9k committed:
Put the EFS mount target in the first private subnet.

All private subnets are currently in the same AZ.

jsf9k committed:
Fix a typo

jsf9k committed:
Create a single EFS mount point in the first private subnet

All the private subnets are currently in the same AZ, so we can't put
one in each private subnet.

jsf9k committed:
Remove extraneous blank line

dav3r commented on locals.tf:
```diff
@@ -35,12 +35,32 @@ locals {
+  # Determine assessment account type based on account name.
+  #
+  # The account name format is "ACCOUNT_NAME (ACCOUNT_TYPE)" - for
+  # example, "Shared Services (Production)".
```
I like how we have been changing this comment in each project to reflect the type of account, so can we change this to:
`# example, "env0 (Production)".`

jsf9k replied:
See commit 60e4037.

dav3r commented on locals.tf:
```diff
@@ -35,12 +35,32 @@ locals {
+  # The Terraform workspace name for this assessment
+  assessment_workspace_name = replace(replace(lower(var.assessment_account_name), "/[()]/", ""), " ", "-")
+
+  assessment_account_name_base = split(" ", var.assessment_account_name)[0]
```
We should probably add a comment here that mentions the assumption that our assessment account names are a single word followed by the account type, e.g. "env0 (Production)".

jsf9k replied:
See commit b1dc956.

jsf9k replied:
See also commit 10cb0a2, which makes the assumption only that there not be a second "(" in the name.

dav3r commented on locals.tf:
```diff
@@ -35,12 +35,32 @@ locals {
   cool_shared_services_cidr_block = data.terraform_remote_state.sharedservices_networking.outputs.vpc.cidr_block
 
-  guacamole_fqdn = format("guac.%s.%s", var.assessment_account_name, var.cool_domain)
+  guacamole_fqdn = format("guac.%s.%s", local.assessment_account_name_base, var.cool_domain)
```
This works for Production environments, but for non-Production environments, I think we want it to be: 
`format("guac.%s.%s.%s", local.assessment_account_name_base, local. workspace_type, var.cool_domain)`

jsf9k replied:
In the case of staging, for example, `var.cool_domain` is `staging.cool.cyber.dhs.gov`.  The change you suggested isn't necessary.

dav3r replied:
Yup, I realized this 2 seconds after I posted it and was just about to confirm that.  Thanks for clarifying.

dav3r reviewed (changes requested):
I have ~~one change that is necessary (unless I'm misunderstanding something) and~~ a couple of comment-related requests.  Take a look and let me know what you think.

jsf9k committed:
Change example to be specific to the account in question

jsf9k committed:
Add a comment pointing out an assumption

jsf9k committed:
Improve code so that its assumption becomes less likely to be false

Previously this line assumed that the assessment account name (module
the "(Staging)" or "(Production)") not contain a space.  Not it only
assumes that it does not contain a "(" character, which is much less
likely.

dav3r commented on locals.tf:
```diff
@@ -55,8 +55,8 @@ locals {
-  # not contain a space character.
-  assessment_account_name_base = split(" ", var.assessment_account_name)[0]
+  # not contain a "(" character.
+  assessment_account_name_base = trimspace(split("(", var.assessment_account_name)[0])
```
Good thinking!

felddy reviewed (approved):
I'm not sure I understand everything I'm reading... but the stuff I did understand blew my socks off! 🌋 
I think we'll need a walk-through of EFS and TG so we're all edumacated.

jsf9k committed:
Add missing ACL rule

Now that EFS is being used, the private subnets need to be able to
egress to the operations subnet via ephemeral ports.

Note that this also required some renumbering of ACL rule numbers.

jsf9k committed:
Fix bug/typo in fstab entry

I did not append ":/" to the EFS ID.

jsf9k committed:
Add mount point via cloud-config instead of using a script
